sudoku_board.h

```diff
@@ -0,0 +1,40 @@
+#pragma
+#include <vector>
+#include <set>
+#include <iostream>
+#include <memory>
+#include <algorithm>
+
+
+enum STATUS_OPTION {WIN, LOSE, PROG};
+
+class SudokuBoard {
+private:
+    int m_size;
```

Comment:
What is the board size? It's length? It's area? Add a doc or choose a better name (I will understand it in a moment from reading the code, but I should be able to understand it now)

Reply:
Also, in any case the board size is an unsigned integer. Usually in order to store size of stuff we use the type `size_t`, to indicate it is a size. This is true for indices of arrays as well.